internal/logging/sentry.go

```diff
@@ -12,7 +12,16 @@ import (
 // sentryWriter creates a zerolog writer for sentry.
 // Uses github.com/archdx/zerolog-sentry which is very simple wrapper.
 func sentryWriter(dsn string) (io.Writer, func(), error) {
-	wr, err := sentrywriter.New(dsn)
+	var opts []sentrywriter.WriterOption
+	if config.Sentry.Debug {
+		opts = append(opts, sentrywriter.WithDebug())
+	}
+	opts = append(opts, sentrywriter.WithSampleRate(config.Sentry.SampleRate))
+	// TODO this calls sentry.Init again (it is initialized in our telemetry package)
+	// We must be able to initialize it only once and pass it into here, or, create it
+	// here with tracing and sample rate enabled and use it also for otel.
+	// Currently it is not possible to have sentry for both errors and tracing.
+	wr, err := sentrywriter.New(dsn, opts...)
```

Comment:
Here we need to contribute new opt functions WithTracingEnabled and I think this is all. Then we would enforce that sentry logging is required if you want also to do tracing to make sure it is initialized before we configure otel.